### PR TITLE
[rubysrc2cpg] builtin functions and types

### PR DESCRIPTION
**Pull request by pandurangpatil**
1. In continuation with this PR - https://github.com/joernio/joern/pull/3066 copied the builtin functions and types from this PR.
2. Before the method name lookup is done inside builtins. Did the lookup inside methods defined inside the file.

### CODE REVIEW

**Comment by pandurangpatil** [comment]
@xavierpinho Copied the changes from this PR https://github.com/joernio/joern/pull/3066 and fixed the failing unit tests

File: joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/astcreation/AstCreator.scala [hunk 1]
```diff
@@ -838,6 +838,7 @@ class AstCreator(
     val methodFullName = packageContext.packageTable
       .getMethodFullNameUsingName(packageStack.toList, name)
       .headOption match {
+      case None if methodNames.contains(name) => methodNames.get(name).getOrElse(DynamicCallUnknownFullName)
```
Comment (DavidBakerEffendi) [inline joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/astcreation/AstCreator.scala:841]:
```suggestion
      case None if methodNames.contains(name) => methodNames(name)
```

**Review by DavidBakerEffendi — APPROVED** [review]
One small change but otherwise LGTM

File: joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/astcreation/AstCreator.scala (hunk 1 above)
Comment (pandurangpatil) [reply joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/astcreation/AstCreator.scala]: done

**Pull request merged**